sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -143,6 +143,17 @@ object GpuDeviceManager extends Logging {
     singletonMemoryInitialized = Errored
     RapidsBufferCatalog.close()
     GpuShuffleEnv.shutdown()
+    // try to avoid segfault on RMM shutdown
+    val timeout = System.nanoTime() + TimeUnit.SECONDS.toNanos(10)
```

Comment:
since this now may potentially block for 10 seconds, we should at least announce it in a log.